pennylane/qnode.py

```diff
@@ -641,10 +641,11 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches
 
             # Autograd or tensorflow: they do not support tuple return with backpropagation
             backprop = False
-            if not isinstance(
-                self._qfunc_output, qml.measurements.MeasurementProcess
-            ) and self.interface in ("tf", "autograd"):
-                backprop = any(qml.math.in_backprop(x) for x in res)
+            if not isinstance(self._qfunc_output, qml.measurements.MeasurementProcess):
+                if self.interface == "autograd":
+                    backprop = any(qml.math.in_backprop(x) for x in res)
+                elif self.interface == "tf":

```

Comment:
Good question 🤔. It doesn't look like a special case for TensorFlow backprop is necessary as we had thought. I removed the case and all the tests still pass. What do you think?

Reply:
Yes here the question is do we want to remove the internal stacking for backpropagation? Because you changed the tests to do the stacking on the user side. It sounds reasonable for me to remove it and have same behavior between backprop and gradient transform. But it should be done in another PR and also for autograd in my opinion. What do you think? @antalszava @eddddddy

Reply:
Reverted 👍